engine.py

```diff
@@ -611,6 +611,43 @@ def _set_project(self):
         self.logger.info("Setting Maya project to '%s'", proj_path)
         pm.mel.setProject(proj_path)
 
+    def get_session_path(self):
+        """
+        Returns the path to the current scene file if it resides on
+        disk. If unsaved, it returns an empty string.
+
+        :returns: Path to the current scene
+        """
+        path = cmds.file(query=True, sceneName=True)
+
+        if isinstance(path, unicode):
+            path = path.encode("utf-8")
+
+        return path
+
+    def get_session_dependencies(self):
```

Comment:
This does not reflect the core interface - a `session=None` parameter is missing!